sdk/python/kfp/compiler/compiler.py

```diff
@@ -689,7 +689,8 @@ def _create_pipeline_workflow(self, parameter_defaults, pipeline, op_transformer
         exit_handler = first_group.exit_op
 
     # The whole pipeline workflow
-    pipeline_name = pipeline.name or 'Pipeline'
+    # It must valid as a subdomain
+    pipeline_name = pipeline.name or 'pipeline'
```

Comment:
IMHO another way to fix this issue with less change to the current behavior is to first call `sanitize_k8s_name` to `pipeline.name` and if that's empty, we replace it with 'Pipeline'. 

WDYT?

Reply:
It is the current behavior. The string object assigned to `pipeline.name` has already been sanitized at [this line](https://github.com/kubeflow/pipelines/blob/3d40bba9a3e14b170a2c05e41a0b0bee64d399b8/sdk/python/kfp/compiler/compiler.py#L832). (https://github.com/kubeflow/pipelines/pull/5120#discussion_r573462700 describes the journey of the string. Please let me know if I missed anything.)

In addition, the `sanitize_k8s_name` returns an empty string if the given string only contains non-ascii char. So the default value is used only when the name of pipeline doesn't contain ascii char at all. I believe this change will never affect to at least English developers. 😉 

![image](https://user-images.githubusercontent.com/96157/108351004-42359c00-7228-11eb-8594-64b3191612af.png)

Reply:
Ah got it.